### PR TITLE
DI-2161 Use multiple docker images to build binary and run executable

### PR DESCRIPTION
**Pull request by chaicesan**
We could make use of the makefile but I'm confident that it is just going to create a load of subshell issues which I want to avoid until we get a working build process.

### CODE REVIEW

**Review by jward-bw — APPROVED** [review]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -10,14 +10,21 @@ def buildAndDeployDockerImage() {
...
             --build-arg BIN=solr-operator \
+            -t solr-operator-build \
+            -f ./build/Dockerfile.build
+        """
+
+        sh """ docker build \
+            --build-arg BUILD_IMG=solr-operator-build . \
             -t ${imageName}:${buildTag} \
-            -f build/Dockerfile.build .
+            -f ./build/Dockerfile.slim
```
Comment (jward-bw) [inline Jenkinsfile:25]: Oh just noticed, you need ` .` on the end of both of these docker commands

**Review by jward-bw — CHANGES_REQUESTED** [review]
^

File: Jenkinsfile (hunk 1 above)
Comment (chaicesan) [reply Jenkinsfile:25]: The dots are there. It's just further up. Want me to put it on the end for readability?

Comment (jward-bw) [reply Jenkinsfile:25]: Oh that's fine, I thought the dot had to be on the end

**Review by jward-bw — APPROVED** [review]

**Pull request merged**